doc/BACKUP.md

```diff
@@ -0,0 +1,400 @@
+existing hard disk, which might matter to you if you are using an
+everyday-use laptop or desktop).
+
+To recover:
+
+* Re-download the `backup` plugin and install Python 3 and the
+  requirements of `backup`.
+* `/path/to/backup-cli restore file:///path/to/nfs/mount ${LIGHTNINGDIR}`
+
+If your backup destination is a network-mounted filesystem that is in a
+remote location, then even loss of all hardware in one location will allow
+you to still recover your Lightning funds.
+
+However, if instead you are just replicating the database on another
+storage device in a single location, you remain vulnerable to disasters
+like fire or computer confiscation.
+
+PostgreSQL Cluster
+------------------
+
+`lightningd` may also be compiled with PostgreSQL support.
+PostgreSQL is generally faster than SQLITE3, and also supports running a
+PostgreSQL cluster to be used by `lightningd`, with automatic replication
+and turnover in case an entire node of the PostgreSQL cluster fails.
+
+Setting this up, however, is more involved.
+
+By default, `lightningd` compiles without PostgreSQL support.
+To enable it, you have to install a developer version of `libpq`.
+On most Debian-derived systems that would be `libpq-dev`.
+To verify you have it properly installed on your system, check if the
+following command gives you a path:
+
+    pg_config --includedir
+
+Versioning may also matter to you.
+For example, Debian Stable ("buster") as of late 2020 provides PostgreSQL 11.9
+for the `libpq-dev` package, but Ubuntu LTS ("focal") of 2020 provides
+PostgreSQL 12.5.
+Debian Testing ("bullseye") uses PostgreSQL 13.0 as of this writing.
+PostgreSQL 12 had a non-trivial change in the way the restore operation is
+done for replication.
+You should use the same PostgreSQL version of `libpq-dev` as what you run
+on your cluster, which probably means running the same distribution on
+your cluster.
+
+You then have to compile `lightningd` with PostgreSQL support.
+
+* Clone or untar a new source tree for `lightning` and `cd` into it.
+  * You *could* just use `make clean` on an existing one, but for the
+    avoidance of doubt (and potential bugs in our `Makefile` cleanup rules),
+    just create a fresh source tree.
+* `./configure`
+  * Add any options to `configure` that you normally use as well.
+* Double-check the `config.vars` file contains `HAVE_POSTGRES=1`.
+  * `grep 'HAVE_POSTGRES' config.vars`
+* `make`
+* If you install `lightningd`, `sudo make install`.
+
+If you were not using PostgreSQL before but have compiled and used
+`lightningd` on your system, the resulting `lightningd` will still
+continue supporting and using your current SQLITE3 database;
+it just gains the option to use a PostgreSQL database as well.
+
+To set up a cluster for a brand new node, follow this (external)
+[guide by @gabridome][gabridomeguide].
+
+[gabridomeguide]: https://github.com/gabridome/docs/blob/master/c-lightning_with_postgresql_reliability.md
+
```

Comment:
The guide is not maintained as the versions of Postgres evolve. Please always check also the Postgres Guide about synchronous replication for your specific version.